tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedScheduledExecutorService/resourcedef/ManagedScheduledExecutorDefinitionOnEJBServlet.java

```diff
@@ -0,0 +1,245 @@
+     */
+    public void testManagedScheduledExecutorDefinitionDefaults_EJB() throws Throwable {
+    	ManagedScheduledExecutorService executor = (ManagedScheduledExecutorService)bean.doLookup("java:comp/concurrent/EJBScheduledExecutorC");
+
+    	CountDownLatch blocker = new CountDownLatch(1);
+    	CountDownLatch allTasksRunning = new CountDownLatch(4);
+
+    	Callable<Integer> txCallable = () -> {
+    		allTasksRunning.countDown();
+    		UserTransaction tx = InitialContext.doLookup("java:comp/UserTransaction");
+    		int initialStatus = tx.getStatus();
+    		tx.begin();
+    		try {
+    			blocker.await(MAX_WAIT_SECONDS * 5, TimeUnit.SECONDS);
+    		} finally {
+    			tx.rollback();
+    		}
+    		return initialStatus;
+    	};
+
+    	Function<String, Object> lookupFunction = jndiName -> {
+    		allTasksRunning.countDown();
+    		try {
+    			blocker.await(MAX_WAIT_SECONDS * 5, TimeUnit.SECONDS);
+    			return (ManagedScheduledExecutorService)bean.doLookup(jndiName);
+    		} catch (InterruptedException | NamingException x) {
+    			throw new CompletionException(x);
+    		}
+    	};
+
+    	try {
+    		Future<Integer> txFuture1 = executor.submit(txCallable);
+
+    		Future<Integer> txFuture2 = executor.submit(txCallable);
+
+    		CompletableFuture<?> lookupFuture1 = executor.completedFuture("java:comp/concurrent/EJBScheduledExecutorC")
+    				.thenApplyAsync(lookupFunction);
+
+    		CompletableFuture<?> lookupFuture2 = executor.completedFuture("java:module/concurrent/ScheduledExecutorB")
+    				.thenApplyAsync(lookupFunction);
+
+    		assertTrue(allTasksRunning.await(MAX_WAIT_SECONDS, TimeUnit.SECONDS),
+    				"ManagedScheduledExecutorService without maxAsync must be able to run async tasks concurrently.");
+
+    		blocker.countDown();
+
+    		int status;
+    		status = txFuture1.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS);
+    		assertEquals(status, Status.STATUS_NO_TRANSACTION, 
+    				"Transaction context must be cleared from first async Callable task " +
+    				"per java:comp/concurrent/EJBScheduledExecutorC configuration.");
+
+    		status = txFuture2.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS);
+    		assertEquals(status, Status.STATUS_NO_TRANSACTION, 
+    				"Transaction context must be cleared from second async Callable task " +
+    				"per java:comp/concurrent/EJBScheduledExecutorB configuration.");
+
+    		assertTrue(lookupFuture1.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS) instanceof ManagedScheduledExecutorService,
+    				"Application context must be propagated to first async Function " +
+    				"per java:comp/concurrent/EJBScheduledExecutorC configuration.");
+
+    		assertTrue(lookupFuture2.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS) instanceof ManagedScheduledExecutorService,
+    				"Application context must be propagated to second async Function " +
+    				"per java:comp/concurrent/ScheduledExecutorB configuration.");
```

Comment:
Typo in error message. Should be EJBScheduledExecutorC not B

Reply:
Fixed.